### DISCUSSION TITLE
Problem with duplicate objects in an array?

### DISCUSSION THREAD

**Discussion by slzatz** [nlohmann/json#2402] [Q&A] [answered]
I am new to nlohmann::json and not a json expert but it appears that duplicate objects in an array are not being handled correctly (or I am doing something wrong).

```
json j = {{"key",{{{"a","b"}, {"a","b"}}}}};
std::cout << j["key"] << std::endl;
std::cout << j.dump() << std::endl;
```
And the output is:

```
[{"a":"b"}]
{"key":[{"a":"b"}]}
```
Produces the expected output if the objects in the array are different.

[compiler explorer gcc trunk nlohmann::json 3.6.0]

**Comment by nlohmann** [comment, marked as answer]
With the outer pair of braces, you create an array. With the next pair of braces, you create either an array or an object, depending on the further entries. In case of pairs where the first entry is a string, an object is created. Hence, you are trying to create the JSON value

```json
[ {"a": "b", "a": "b"} ]
```

As there are no duplicate keys in JSON objects, this boils down to the value

```json
[ {"a": "b"} ]
```

that you observe. See the example in https://nlohmann.github.io/json/api/basic_json/array/ for an example how to make a list of pairs.

**Comment by slzatz** [comment]
Thanks for the quick response and for a really terrific and useful library.
